module1/src/test/java/net/broscorp/primitives/PrimitivesTest.java

```diff
@@ -0,0 +1,50 @@
+package net.broscorp.primitives;
+
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import org.junit.jupiter.api.Test;
+
+public class PrimitivesTest {
+
+  @Test
+  public void overflowTest() {
+
+    //variable is in positive range
+    int testedVariable = 1;
+    //adding positive number
+    testedVariable += Integer.MAX_VALUE;
+
+    //for some reason variable is now negative(((
+    assertTrue(testedVariable < 0);
+
+  }
+
+  @Test
+  public void failCastTest() {
+    //long max value is far more than int max value
+    //this operation results in overflow
+    long extremelyBigLong = Long.MAX_VALUE;
+    assertNotEquals(extremelyBigLong, (int) extremelyBigLong);
+
+    //double stores the number the different way compared to float
+    float pi = 3.141592F;
+    assertNotEquals(Float.toString(pi), Double.toString(pi));
+
+  }
```

Comment:
Add tests to illustrate upcasting and downcasting of floating point types without using wrapper classes.

Reply:
I don't think that upcasting and downcasting is a right term to use in this case. Also, i think that casting double to float without converting it to string would not be that illustrative due to compiler optimizations. I'll add them though.